pf/tests/provider_read_test.go

```diff
@@ -122,32 +122,61 @@ func TestReadFromRefresh(t *testing.T) {
 	      "special": true,
 	      "upper": true
 	    },
-            "inputs": {}
+	    "inputs": {
+	      "length": 8,
+	      "lower": true,
+	      "minLower": 0,
+	      "minNumeric": 0,
+	      "minSpecial": 0,
+	      "minUpper": 0,
+	      "number": true,
+	      "overrideSpecial": "_%@:",
+	      "special": true,
+	      "upper": true
+	    }
 	  }
 	}]`
 
-	// TODO populate inputs
-	// "inputs": {
-	//   "__defaults": [
-	// 	"lower",
-	// 	"minLower",
-	// 	"minNumeric",
-	// 	"minSpecial",
-	// 	"minUpper",
-	// 	"number",
-	// 	"upper"
-	//   ],
-	//   "length": 8,
-	//   "lower": true,
-	//   "minLower": 0,
-	//   "minNumeric": 0,
-	//   "minSpecial": 0,
-	//   "minUpper": 0,
-	//   "number": true,
-	//   "overrideSpecial": "_%@:",
-	//   "special": true,
-	//   "upper": true
-	// }
-
 	testutils.ReplaySequence(t, server, testCase)
 }
+
+func TestImportRandomPassword(t *testing.T) {
+	server := newProviderServer(t, testprovider.RandomProvider())
+	testCase := `
+	{
+	  "method": "/pulumirpc.ResourceProvider/Read",
+	  "request": {
+	    "id": "supersecret",
+	    "urn": "urn:pulumi:v2::re::random:index/randomPassword:RandomPassword::newPassword",
+	    "properties": {}
+	  },
+	  "response": {
+	    "id": "none",
```

Comment:
Does this need to be a random value to have a unique URN?

Reply:
I don't think so it doesn't participate in URN that's computed by `random:index/randomPassword:RandomPassword newPassword` parameters. Perhaps another q @Frassle can doublecheck with us. The "none" here is returned straight from Plugin Framework provider.